### ISSUE TITLE
Set a Form Field Value Through the URL

### ISSUE THREAD

**Issue by bradlis7** [getgrav/grav#826]
I am using the Form and Email plugins to have a contact form, but I'd like to have buttons on my site in certain places that automatically fill in the subject based on what they click. A [Stack Overflow answer](https://stackoverflow.com/a/15606977/179311) suggests to symfony users to access `{{ app.request }} `, but it doesn't seem to be available to me.

I would be fine with setting a default value in the `.md` file, processing it in the twig template file, or have it work automatically for me. It seems like the url similar to how taxonomy filters would work (`/contact/subject:Button+A`), if that would make sense to the grav team, but `/contact?subject=Button+A` is perfectly fine as well.

---

I was told by @flaviocopes on gitter that this is currently not supported, and to open an issue.

**Comment by rhukster** [comment]
Just FYI, the stack overflow example is specific to Symfony framework, hence the `{{ app.request }}`.  In grav you would get to a specific request query via `{{ uri.query('var') }}` or `{{ uri.query }}` for all the query strings as an array.

Now that support in the Forms that generate the contact form is probably missing.

**Comment by bradlis7** [comment]
Thanks @rhukster! At the top of my theme file `templates/forms/fields/text/text.html.twig`, I set the value with this command:

    {% set value = (value ?? uri.query(field.name)) ?? field.default %}

It would be a nice addition if I could set the default in my `form.md` as an alternative to those that don't want to have to customize each field:

          default: "{{ uri.query('subject') }}"

It looks like that field may not get processed as twig template text though.

**Comment by flaviocopes** [comment]
In the just released [beta](https://github.com/getgrav/grav-plugin-form/releases/tag/1.3.0-beta.5 ) of the plugin, you can do in your theme's `templates/forms/field.html.twig`

```
{% extends "forms/default/field.html.twig" %}
{% block field %}
    {% set value = evaluate(value) %}
    {{ parent() }}
{% endblock %}
```

and will evaluate the default property. You set it using `default: "uri.param('subject')"`

**Comment by hugoaf** [comment]
Hi Flavio, sorry to comment on a closed issue, but I think is not working any more.

Thanks to your code, I have this working ok in a site running:

Grav v1.1.15 
 Admin Panel  v1.2.10 
 Data Manager  v1.0.7
 Email  v2.5.3
 Error  v1.6.0
 Form  v2.0.10 
 Gantry 5 Framework v5.4.6 
 Login  v2.2.1 
 Markdown Notices  v1.0.0
 Problems  v1.4.5 

Today I did a copy of this website and update to newest versions:

Grav v1.2.4 
 Admin Panel  v1.4.2
 Data Manager  v1.0.7
 Email  v2.5.3
 Error  v1.6.0
 Form  v2.6.0
 Form Database v1.0.0
 Gantry 5 Framework v5.4.12
 Login  v2.3.0
 Markdown Notices  v1.0.0
 Problems  v1.4.6

But unfortunately it quits working, no errors in php log, all I got is this text in front end (similar for all pages):

```
partials/page_head.html.twig @nucleus/layout/offcanvas.html.twig
@nucleus/layout/section.html.twig @nucleus/layout/section.html.twig @nucleus/layout/section.html.twig @nucleus/layout/container.html.twig @nucleus/layout/section.html.twig
```

Removing `{% set value = evaluate(value) %}` makes my web site back to life, but of course without the url param to form field default feature.
 
Thanks,
Hugo.
